app/src/main/cpp/ControllerContainer.cpp

```diff
@@ -188,6 +189,64 @@ ControllerContainer::InitializeBeam() {
   }
 }
 
+void ControllerContainer::SetHandJointLocations(const int32_t aControllerIndex, std::vector<vrb::Matrix>& jointTransforms)
+{
+    if (!m.Contains(aControllerIndex))
+        return;
+
+    Controller &controller = m.list[aControllerIndex];
+    if (!controller.beamParent)
+        return;
+
+    // Initialize hand joints if needed
+    if (controller.handJointTransforms.size() == 0) {
+        controller.handJointTransforms.resize(jointTransforms.size());
+
+        CreationContextPtr create = m.context.lock();
+        controller.handMeshToggle = Toggle::Create(create);
+        controller.beamParent->AddNode(controller.handMeshToggle);
+
+        ProgramPtr program = create->GetProgramFactory()->CreateProgram(create, 0);
+        RenderStatePtr state = RenderState::Create(create);
+        state->SetProgram(program);
+        vrb::Color handColor = controller.leftHanded ?
+                vrb::Color(0.0f, 1.0f, 0.0f) : vrb::Color(1.0f, 1.0f, 0.0f);
```

Comment:
Not totally convinced about having different colors for the hands. Do you have any other real world example using the same approach?

Reply:
I think the same. It is ok for debugging but eventually I think it's more natural to paint both hands in the same color.

Reply:
I don't know of other examples, it just seemed useful to use different colors to better differentiate which ball belongs to which hand, specially when they are close to each other, which also subtly helps predicting where each ball is gonna move.    

I'm fully open to suggestions here. I can turn both the same color, but which one is best? Maybe the color of the hands can vary based on some specific gesture or condition. There are many things we can do. For now any variant would work for me.

I make both greens, and add some translucency and then re-evaluate.

Reply:
Done in the new version.